File: src/main/java/io/github/dqualizer/dqlang/types/adapter/constants/resilienceTesting/ResilienceTestConstants.java
package io.github.dqualizer.dqlang.types.adapter.constants.resilienceTesting;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.github.dqualizer.dqlang.types.adapter.constants.ResponseMeasure;
import lombok.Getter;
import lombok.ToString;

/**
 * Java class for load test constants definitions
 */
@Getter
@ToString
public class ResilienceTestConstants {

    @JsonProperty("recovery_time")
    private ResponseMeasure recoveryTime;
    @JsonProperty("response_time")
    private ResponseMeasure responseTime;
    @JsonProperty("error_rate")
    private ResponseMeasure errorRate;
}

File: src/main/kotlin/io/github/dqualizer/dqlang/types/adapter/constants/resilienceTest/ResilienceTestConstants.kt
package io.github.dqualizer.dqlang.types.adapter.constants.resilienceTest

import com.fasterxml.jackson.annotation.JsonProperty
import io.github.dqualizer.dqlang.types.adapter.constants.SatisfactionValue

data class ResilienceTestConstants(
  @JsonProperty("recovery_time")
  val recoveryTime: SatisfactionValue? = null,

  @JsonProperty("response_time")
  val responseTime: SatisfactionValue? = null,

  @JsonProperty("error_rate")
  val errorRate: SatisfactionValue? = null
)
